assets/js/modules/adsense/datastore/settings.js

```diff
@@ -0,0 +1,444 @@
+/**
+ * modules/adsense data store: settings.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import {
+	isValidAccountID,
+	isValidClientID,
+} from '../util';
+import { STORE_NAME } from './constants';
+
+const { commonActions, createRegistrySelector, createRegistryControl } = Data;
+
+// Actions
+const SUBMIT_CHANGES = 'SUBMIT_CHANGES';
+const START_SUBMIT_CHANGES = 'START_SUBMIT_CHANGES';
+const FINISH_SUBMIT_CHANGES = 'FINISH_SUBMIT_CHANGES';
+
+const FETCH_SAVE_USE_SNIPPET = 'FETCH_SAVE_USE_SNIPPET';
+const START_FETCH_SAVE_USE_SNIPPET = 'START_FETCH_SAVE_USE_SNIPPET';
+const FINISH_FETCH_SAVE_USE_SNIPPET = 'FINISH_FETCH_SAVE_USE_SNIPPET';
+const CATCH_FETCH_SAVE_USE_SNIPPET = 'CATCH_FETCH_SAVE_USE_SNIPPET';
+
+const RECEIVE_SAVE_USE_SNIPPET = 'RECEIVE_SAVE_USE_SNIPPET';
+
+const COMPLETE_ACCOUNT_SETUP = 'COMPLETE_ACCOUNT_SETUP';
+const COMPLETE_SITE_SETUP = 'COMPLETE_SITE_SETUP';
+
+// The original account status on pageload is a specific requirement for
+// certain parts of the AdSense setup flow.
+const RECEIVE_ORIGINAL_ACCOUNT_STATUS = 'RECEIVE_ORIGINAL_ACCOUNT_STATUS';
+
+export const INITIAL_STATE = {
+	isDoingSubmitChanges: false,
+	isFetchingSaveUseSnippet: false,
+	originalAccountStatus: undefined,
+};
+
+export const actions = {
+	*fetchSaveUseSnippet( useSnippet ) {
+		let response, error;
+
+		if ( 'undefined' === typeof useSnippet ) {
+			error = {
```

Comment:
Hmm not sure what the benefit here is. Since we use plain objects overall, I think it's okay to do that here as well.

Reply:
I haven't seen that but it's not a bad idea since `error` is usually set from the catch block, it will usually be an instance of `Error` rather than a plain object.